spec/tbd_tests_spec.rb

```diff
@@ -1566,3 +1566,414 @@
+              "type": "number",
+              "minimum": 0
+            },
+            "parapet": {
+              "title": "Roof/wall or exposed-floor/wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "fenestration": {
+              "title": "Window, door, skylight perimeter PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "concave": {
+              "title": "Wall corner [0°,135°) PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "convex": {
+              "title": "Wall corner (225°,360°] PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "balcony": {
+              "title": "Floor/balcony edge PSI ",
+              "type": "number",
+              "minimum": 0
+            },
+            "party": {
+              "title": "Party wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "grade": {
+              "title": "Floor/foundation edge PSI",
+              "type": "number",
+              "minimum": 0
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id"
+          ],
+          "minProperties": 2
+        },
+        "KHI": {
+          "description": "KHI-value (in W/K) for a point thermal bridge",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "Unique KHI identifier",
+              "type": "string"
+            },
+            "point": {
+              "title": "Point KHI-value",
+              "type": "number",
+              "minimum": 0
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id",
+            "point"
+          ]
+        },
+        "Edge": {
+          "description": "Surface(s) edge as thermal bridge",
+          "type": "object",
+          "properties": {
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            },
+            "type": {
+              "title": "PSI-set type, e.g. 'parapet'",
+              "type": "string"
+            },
+            "length": {
+              "title": "Edge length (m), 10cm min",
+              "type": "number",
+              "minimum": 0.1
+            },
+            "surfaces": {
+              "title": "Surface(s) connected to edge",
+              "type": "array",
+              "items": {
+                "type": "string"
+              },
+              "minItems": 1,
+              "uniqueItems": true
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "psi",
+            "type",
+            "length",
+            "surfaces"
+          ]
+        },
+        "Surface": {
+          "description": "Surface default PSI-set (optional)",
```

Comment:
I'm confused about whether this Surface object refers to an OS/E+ surface or if it is like a DefaultPSIKHISet?

Reply:
Maybe this [example](https://github.com/rd2/tbd/blob/json/json/tbd_PSI_combo2.json) helps. For starters (or maybe more of a reminder), "psis" (PSI definition) and "khis" (KHI definition) are libraries, analogous to default construction sets in OpenStudio. If one is not interested in the few default sets provided with TBD (e.g. BC Hydro sets), one can use the JSON file (as input) to specify their own default "Unit" PSI-set, which in turn would be used by TBD to initially set all thermal bridges (TBs) in a building (based on enumerated allowable PSI types, e.g. "parapet" vs "fenestration"). One can also specify default Story, Space and/or Surface PSI-sets, overriding the default Unit PSI-set. So for instance, any Surface set will supersede its immediate Space set (if present). An Edge PSI-set is slightly different in that it allows users to hijack a specific edge they suspect will be (or notice is) misdiagnosed by TBD (e.g. "concave" corner rather than a "party" wall edge - it'll happen), or even add an edge that isn't even in the OSM geometry (similar to adding a _point_ TB). So in the cited example, all TBs are by default initially processed using the building Unit PSI-set ("Awesome"), yet the "ok" set shall instead be used for the "ground-floor restaurant South-wall" (overriding the "Awesome set"). This is very similar to default constructions vs surface-specified constructions. Finally, any edge connecting the "ground-floor restaurant West-wall" and the "ground-floor restaurant party wall", and 4.3m in length, will be processed using the "Part wall edge", also overriding the Unit set. If no such edge(s) exist(s) in Topolys/TBD model, then TBD will process it as an additional edge (similarly to a _point_ TB).